src/Tools/dotnet-counters/Exporters/ConsoleWriter.cs

```diff
@@ -150,31 +150,43 @@ public void AssignRowsAndInitializeDisplay()
             {
                 Console.WriteLine($"[{provider.Name}]"); row++;
 
+                foreach (var counter in provider.Counters.Values)
+                {
+                    counter.Row = -1;
+                    if (counter.RenderValueInline)
```

Comment:
I don't understand what is going on here:
We will go into this if statement if either `TagSets.Count == 0 || TagSets.Count == 1 && string.IsNullOrEmpty(TagSets.Keys.First())` So, either TagSets has nothing in it, or it has one thing in it, but the first key is null or empty. If there is nothing in it, then we would not go through the next foreach loop, if it has one thing in it, then we could just set the row = -1 instead of going through a foreach loop.

Reply:
I agree - in my adhoc testing the performance overhead was not noticeable.  For context, `RenderValueInline` was added as part of #2373.  I am matching the row assignment logic further down in the method for `counter.TagSets.Values` enumeration.

Reply:
Does your comment below apply to this code path? "When a new counter is found via the tracevent callback, we reset all of the counter object row numbers to -1 and then recompute them."

Reply:
> Does your comment below apply to this code path? "When a new counter is found via the tracevent callback, we reset all of the counter object row numbers to -1 and then recompute them."

This specific loop resets the row numbers to -1.